pennylane/ops/op_math/sum.py

```diff
@@ -0,0 +1,291 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the Sum class which contains logic for
+computing the sum of operations.
+"""
+from functools import reduce
+
+import numpy as np
+import pennylane as qml
+
+from pennylane import math
+from pennylane.operation import Operator
+
+
+def op_sum(*summands, do_queue=True, id=None):
+    r"""Construct an operator which is the sum of the given operators.
+
+    Args:
+        *summands (tuple[~.operation.Operator]): the operators we want to sum together.
+
+    Keyword Args:
+        do_queue (bool): determines if the sum operator will be queued (currently not supported).
+            Default is True.
+        id (str or None): id for the sum operator. Default is None.
+
+    Returns:
+        ~ops.op_math.Sum: the operator representing the sum of summands.
+
+    ..seealso:: :class:`~.ops.op_math.Sum`
+
+    **Example**
+
+    >>> summed_op = op_sum(qml.PauliX(0), qml.PauliZ(0))
+    >>> summed_op
+    PauliX(wires=[0]) + PauliZ(wires=[0])
+    >>> summed_op.matrix()
+    array([[ 1,  1],
+           [ 1, -1]])
+    """
+    return Sum(*summands, do_queue=do_queue, id=id)
```

Comment:
Since this function simply instantiates the class, this almost begs of the question of --- should we rename the `Sum` class to `sum`? Although it feels wrong to suggest that!

Reply:
It does feel weird as we have the convention to name classes with a capital letter. I think it would still be valuable to use a top level wrapper function called `sum` that wraps the class `Sum`. Not sure if we can get away with having the same name for the wrapper function and the class. 

Keeping the wrapper function would allow us to be flexible with the implementation of the class in the future.